### ISSUE TITLE
ShowMeYourSkin 1.19.2 - 1.6.3 Hanging servers on Shutdown

### ISSUE THREAD

**Issue by ImKringle** [enjarai/show-me-your-skin#58]
Ran into this when troubleshooting a Modpack, this mod by itself on a server causes an abnormal shutdown (An autokill after 60-120 seconds) - Not sure on the exact cause as I don't have the time to check that end, but wanted to let you know there's something funky in there

Server Logs:
https://mclo.gs/1DIEecd
- Added where the Stop command + eventual kill was for context

Just incase Mclogs clears after a while:
[latest (11).log](https://github.com/enjarai/show-me-your-skin/files/12640626/latest.11.log)

1:1 of Server files:
https://cdn.discordapp.com/attachments/1120177535708315722/1152575026059542538/ShowMeYourSkin-Issue.zip

Hope this helps - Cheers!

**Comment by enjarai** [comment]
Thanks for the detailed report, always great to see.

I have some idea of how to fix this, but I haven't gotten around to implementing it yet, though 1.20.2 being around the corner is probably a great opportunity to finally do it. 

It should be noted that I've dropped support for 1.19.2 after 1.20 came out, as it can be quite a burden to maintain multiple versions of the same mod. I can imagine this being quite an annoying issue though, so I'll look into the possibility of a bugfix backport. No promises though, the MC codebase changed quite a lot between 1.19.2 and 1.19.4. (blame mojang lol)

**Comment by ImKringle** [comment]
I kinda figured as 1.19.x is old and it really isnt a *huge* deal - Just found it during setup and figured id note it down incase you had a hunch haha

If it's still happening in the 1.20 release tho, yeah 1.20.2 is a great time xD

**Comment by enjarai** [comment]
Okay, so this bug originates in a common library I use for my mods called CICADA. Normally you wouldn't have to deal with it, as its included in the jar file, and Fabic will load it from there, but that means to fix it properly, I'll have to push a version of this mod that includes a newer version of CICADA, even if no code from this mod changes. However, if you manually provide a never version of an included library to Fabric, it'll load that instead of the included version.

So long story short, this means you should be able to fix it by adding this jar to your mods folder:
https://maven.enjarai.dev/releases/nl/enjarai/cicada-lib/0.4.4/cicada-lib-0.4.4.jar

It's not on any mod publishing websites, cause it usually doesn't need to be, but feel free to check the source [here](https://github.com/enjarai/cicada-lib).

**Comment by enjarai** [comment]
I'll close this issue now, and will post further updates in #38.
